operators/ehr_query_llm/fhir/resource_sanitizer.py

```diff
@@ -0,0 +1,658 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import base64
+import json
+import logging
+import re
+from typing import Any, Dict, List, Optional, Tuple
+
+import pydantic
+from fhir.resources.R4B import FHIRAbstractModel, construct_fhir_element
+from fhir.resources.R4B.allergyintolerance import AllergyIntolerance, AllergyIntoleranceReaction
+from fhir.resources.R4B.attachment import Attachment
+from fhir.resources.R4B.careplan import CarePlan
+from fhir.resources.R4B.coding import Coding
+from fhir.resources.R4B.condition import Condition
+from fhir.resources.R4B.diagnosticreport import DiagnosticReport
+from fhir.resources.R4B.documentreference import DocumentReference, DocumentReferenceContent
+from fhir.resources.R4B.encounter import Encounter
+from fhir.resources.R4B.familymemberhistory import FamilyMemberHistory
+from fhir.resources.R4B.humanname import HumanName
+from fhir.resources.R4B.imagingstudy import ImagingStudy, ImagingStudySeries
+from fhir.resources.R4B.immunization import Immunization
+from fhir.resources.R4B.medication import Medication
+from fhir.resources.R4B.medicationrequest import MedicationRequest
+from fhir.resources.R4B.medicationstatement import MedicationStatement
+from fhir.resources.R4B.observation import Observation, ObservationComponent
+from fhir.resources.R4B.patient import Patient
+from fhir.resources.R4B.procedure import Procedure
+from fhir.resources.R4B.servicerequest import ServiceRequest
+
+logger = logging.getLogger("FHIRResourceSanitizer")
+
+
+class FHIRResourceSanitizer:
+    """
+    A helper class that extracts/transforms a single FHIR document/record.
+    """
+
+    @staticmethod
+    def sanitize(data: Dict) -> Optional[Dict]:
+        """The sanitize method extracts wanted data, removes noises and transforms the data structure
```

Comment:
What are some examples of "noise" to remove?

Reply:
Will change the wording, examples are raw entries missing required attributes.